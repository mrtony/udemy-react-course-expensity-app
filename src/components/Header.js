import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth';

export const Header = ({startLogout}) => (
    <div>
        <h1>Expensity App</h1>
        <div>
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active">Add</NavLink>
        <NavLink to="/help" activeClassName="is-active">Help</NavLink>
        </div>
        <button onClick={startLogout}>Logout</button>
    </div>
);

const mapDispatchToProps = dispatch => (
  {
    startLogout: () => dispatch(startLogout())
  }
);

export default connect(undefined, mapDispatchToProps)(Header);