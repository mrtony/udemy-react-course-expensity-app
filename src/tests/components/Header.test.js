import React from 'react';
//import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

// test('Should render Header correctly using snapshot', () => {
//   //使用jest snapshot的作法
//   const renderer = new ShallowRenderer();
//   renderer.render(<Header />);
//   expect(renderer.getRenderOutput()).toMatchSnapshot();
// });

describe('Header component', () => {
  test('至少要有1個<h1>tag using enzyme', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('h1').length).toBe(1);
  });

  test('<h1>tag文字為Expensity App - using enzyme', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('h1').text()).toBe('Expensity App');
  });
  
  test('snapshot test - using enzyme', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
})




