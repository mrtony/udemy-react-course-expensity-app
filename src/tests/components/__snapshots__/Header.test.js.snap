// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component snapshot test - using enzyme 1`] = `
<div>
  <h1>
    Expensity App
  </h1>
  <div>
    <NavLink
      activeClassName="is-active"
      aria-current="page"
      exact={true}
      to="/"
    >
      Dashboard
    </NavLink>
    <NavLink
      activeClassName="is-active"
      aria-current="page"
      to="/create"
    >
      Add
    </NavLink>
    <NavLink
      activeClassName="is-active"
      aria-current="page"
      to="/help"
    >
      Help
    </NavLink>
  </div>
</div>
`;
